components/state/state.js

```diff
@@ -0,0 +1,15 @@
+import { fetchCharacters } from "../../index.js";
+
+export let state = {
+  maxPage: "",
+  page: 1,
+  searchQuery: "",
+};
+
+export function setState(newState) {
+  if (newState !== undefined) {
+    state = newState;
+  }
+
+  fetchCharacters(state);
+}
```

Comment:
WTF!? :D 
das musst du mir morgen nochmal erklären :)

Reply:
It would be a pleasure for me😂